flutter/lib/src/integrations/native_app_start_integration.dart

```diff
@@ -1,256 +1,57 @@
+      _appStartEndCompleter.complete();
     }
-    return _appStartCompleter.future
-        .timeout(_timeoutDuration, onTimeout: () => null);
   }
 
-  @visibleForTesting
-  static void clearAppStartInfo() {
-    _appStartInfo = null;
-    _appStartCompleter = Completer<AppStartInfo?>();
-    didAddAppStartMeasurement = false;
-  }
-
-  /// Reset state
-  @visibleForTesting
-  static void reset() {
-    appStartEnd = null;
-    _didFetchAppStart = false;
-  }
+  final Completer<void> _appStartEndCompleter = Completer<void>();
 
   @override
-  void call(Hub hub, SentryFlutterOptions options) {
-    if (isIntegrationTest) {
-      final appStartInfo = AppStartInfo(
-        AppStartType.cold,
-        start: DateTime.now(),
-        end: DateTime.now().add(const Duration(milliseconds: 100)),
-        pluginRegistration:
-            DateTime.now().add(const Duration(milliseconds: 50)),
-        sentrySetupStart: DateTime.now().add(const Duration(milliseconds: 60)),
-        nativeSpanTimes: [],
-      );
-      setAppStartInfo(appStartInfo);
-      return;
-    }
-
-    if (_didFetchAppStart) {
-      return;
-    }
-
+  void call(Hub hub, SentryFlutterOptions options) async {
     _frameCallbackHandler.addPostFrameCallback((timeStamp) async {
-      _didFetchAppStart = true;
-      final nativeAppStart = await _native.fetchNativeAppStart();
-      if (nativeAppStart == null) {
-        setAppStartInfo(null);
-        return;
-      }
-
-      final sentrySetupStartDateTime = SentryFlutter.sentrySetupStartTime;
-      if (sentrySetupStartDateTime == null) {
-        setAppStartInfo(null);
-        return;
-      }
-
-      final appStartDateTime = DateTime.fromMillisecondsSinceEpoch(
-          nativeAppStart.appStartTime.toInt());
-      final pluginRegistrationDateTime = DateTime.fromMillisecondsSinceEpoch(
-          nativeAppStart.pluginRegistrationTime);
-
-      if (options.autoAppStart) {
-        // We only assign the current time if it's not already set - this is useful in tests
-        appStartEnd ??= options.clock();
-
-        final duration = appStartEnd?.difference(appStartDateTime);
-
-        // We filter out app start more than 60s.
-        // This could be due to many different reasons.
-        // If you do the manual init and init the SDK too late and it does not
-        // compute the app start end in the very first Screen.
-        // If the process starts but the App isn't in the foreground.
-        // If the system forked the process earlier to accelerate the app start.
-        // And some unknown reasons that could not be reproduced.
-        // We've seen app starts with hours, days and even months.
-        if (duration != null && duration.inMilliseconds > _maxAppStartMillis) {
-          setAppStartInfo(null);
-          return;
+      try {
+        if (!options.autoAppStart && _appStartEnd == null) {
+          await _appStartEndCompleter.future
+              .timeout(const Duration(seconds: 10));
```

Comment:
this means that users who want to manually set an app start cannot do so anymore if the postFrameCallback is not triggered for example if they initialize the sdk in an isolate.

we should explain this in the **changelog**

Reply:
I'm not sure here if it's fine to release this in `8.10.0` and add it to the changelog that this is a breaking change for app start

Reply:
on the other hand the amount of users that do this should be minimal. 

cc @stefanosiano any thoughts on this?

Reply:
Having this outside of `postFrameCallback` blocks the UI when awaiting the completer, so users are never able to even call `setAppStartEnd`.

What we could do is not call the it if we are not in the main isolate. wdyt?

Reply:
is it possible for it to trigger when not on a main isolate?

and would that still fix this issue https://github.com/getsentry/sentry-dart/issues/2103

Reply:
So I added a test, and it would work, but we have an issue we had before. While we can use `Isolate.current.debugName` to check for `main` isolate, nothing in the documentation indicates that this is guaranteed to work.

Maybe we should document that, for now, this integration only works with sentry initialised on the main isolate. 

Technically it would not be breaking, as the previous implementation also relied upon the postFrameCallback and was broken anyway. We ran in the completer timeout and users could not manually set app start end anyway.

Also, whats the point of doing a ui transaction if we are not on the main isolate? All UI work is done on main, so per definition there should be no ui transactions on other isolates, no? 🤔

Reply:
Let's keep app starts for main isolate and it would just not be reported if the SDK was initialized in another isolate.